src/pageNavigation.js

```diff
@@ -147,14 +147,33 @@ async function addListener(listener, {
 
     // Compile the match patterns for the listener
     const matchPatternSet = matching.createMatchPatternSet(matchPatterns);
-    // Register a content script for the listener
-    const contentScript = await browser.contentScripts.register({
-        matches: matchPatterns,
-        js: [{
-            file: pageNavigationContentScript
-        }],
-        runAt: "document_start"
-    });
+    // Only Firefox supports dynamic content script loading at this time.
+    let contentScript;
+    try {
+        // getBrowserInfo is Firefox-only as well.
+        const browserInfo = await browser.runtime.getBrowserInfo();
```

Comment:
This is just the same check as above to try to detect the browser and handle dynamic content script registration for Firefox, or log a warning for Chrome.